### ISSUE TITLE
RLENGTH is wrong for regexes with two identical consecutive '?' matches

### ISSUE THREAD

**Issue by pstumpf** [ThomasDickey/original-mawk#76]
Consider this testcase:
```
match($1, /foobar1?1?/) {
        print RLENGTH
}
```
```
$ echo foobar11 | mawk -f test.awk 
7
```
bwk's `awk`, `gawk`  as well as `mawk` when compiled with system regex instead of the internal regex engine print the correct result `8`.

**Comment by ThomasDickey** [comment]
thanks - I can reproduce this.

**Comment by ThomasDickey** [comment]
fixed in [20240819](https://invisible-island.net/mawk/CHANGES.html#t20240819)
